mlflow/environment_variables.py

```diff
@@ -622,7 +622,9 @@ def get(self):
 MLFLOW_HTTP_POOL_MAXSIZE = _EnvironmentVariable("MLFLOW_HTTP_POOL_MAXSIZE", int, 10)
 
 
-#: Whether to enable Databricks SDK. If true, MLflow uses databricks-sdk API to send HTTP requests
-# to Databricks endpoint, otherwise MLflow uses ``requests`` library to send HTTP requests
-# to Databricks endpoint
-MLFLOW_ENABLE_DATABRICKS_SDK = _BooleanEnvironmentVariable("MLFLOW_ENABLE_DATABRICKS_SDK", True)
+#: Whether to disable Databricks SDK. If False, MLflow uses databricks-sdk API to send HTTP requests
+#: to Databricks endpoint, otherwise MLflow uses ``requests`` library to send HTTP requests
+#: to Databricks endpoint. Note that if you want to use OAuth authentication, you have to
+#: enable Databricks SDK.
+#: (default: ``False``)
+MLFLOW_DISABLE_DB_SDK = _BooleanEnvironmentVariable("MLFLOW_DISABLE_DB_SDK", False)
```

Comment:
Is there any reason we switch from `ENABLE` to `DISABLE`? I think the original one is good, "not disable" is less intuitive vs "enable".

Reply:
oh it is suggested by https://databricks.slack.com/archives/C01L70G65GD/p1717760687596869?thread_ts=1717756439.911229&cid=C01L70G65GD 
i am ok with either enable or disable

Reply:
Oh I thought the suggestion is about default value (so users can use OAuth by default) rather than naming, but yea we can wait for her comment as well.

Reply:
I'm fine with both either! My original thought was that if less users want to opt-out this behavior, then only they need to set this env var, and in such case it will be more intuitive if they use `MLFLOW_DISABLE_DB_SDK=True` than `MLFLOW_ENABLE_DB_SDK=False`. But if more users use this env var I would prefer `MLFLOW_ENABLE_DB_SDK` as a general name.

Reply:
We might have many user want to use this config, then `MLFLOW_ENABLE_DB_SDK` sounds better